### PR TITLE
[#2980] separate file extension an daggregation matching logic

### PR DESCRIPTION
**Pull request by sblack-usu**
### Pull Request Checklist: 
- [x] Positive Test Case Written by Dev

- [x] Automated Testing

- [x] Sufficient User and Developer Documentation

- [x] Passing Jenkins Build

- [x] Peer Code review and approval

### Positive Test Case
1. Setup a webapp for a single file aggregation (.refts.json).  Add a launching pattern on both the file level and aggregation level.  for the file extension the aggregation's extension (.json)
2. Setup a webapp for a multi file aggregation (.nc).  Add a launching pattern on both the file level and aggregation level.  for the file extension, use the main file extension (.nc)
3. Add both webapps to your open with list
4. Create a composite resource with the aggregation types the two web apps were created with.
5. Right click on the multi file aggregation, verify the correct web app shows and launches correctly.  Open the multi file aggregation folder, right click on the files inside and verify the aggregation level web app doesn't show.  Verify the file extension launching pattern shows.
6. Right click on the single file aggregation.  Verify both aggregation level and file extension aggregations show and launch correctly.

### CODE REVIEW

File: theme/static/js/hs_file_browser.js [hunk 1]
```diff
@@ -596,19 +596,23 @@ function bindFileBrowserItemEvents() {
             }
             menu = $("#right-click-menu");
 
-            var fileAggType = $(event.target).closest("li").find("span.fb-logical-file-type").attr("data-logical-file-type");
+            fileAggType = [];
+            // main-file is available on the aggregation folder and only single file aggregations have a data-pk of 1 on the file
+            if ($(event.target).closest("li").attr("main-file") || $(event.target).closest("li").attr("data-pk") === "1"){
```
Comment (sblack-usu) [inline theme/static/js/hs_file_browser.js]: @pkdash - I kind of hate this logic, do you have any better suggestions?

File: theme/static/js/hs_file_browser.js [hunk 2]
```diff
@@ -596,19 +596,23 @@ function bindFileBrowserItemEvents() {
             }
             menu = $("#right-click-menu");
 
-            var fileAggType = $(event.target).closest("li").find("span.fb-logical-file-type").attr("data-logical-file-type");
+            fileAggType = [];
```
Comment (hyi) [inline theme/static/js/hs_file_browser.js]: ```var``` should be added for ```fileAggType``` here when it is first initialized here, right?

File: theme/static/js/hs_file_browser.js [hunk 3]
```diff
@@ -596,19 +596,23 @@ function bindFileBrowserItemEvents() {
...
+            // main-file is available on the aggregation folder and only single file aggregations have a data-pk of 1 on the file
+            if ($(event.target).closest("li").attr("main-file") || $(event.target).closest("li").attr("data-pk") === "1"){
+                var fileAggType = $(event.target).closest("li").find("span.fb-logical-file-type").attr("data-logical-file-type");
+            }
```
Comment (hyi) [inline theme/static/js/hs_file_browser.js]: If ```fileAggType``` is initialized before, see my comment above, it does not have to be redefined here and can just be assigned a value here, and I assume the returned value should be a array type to conform with the variable initialization type.

File: theme/static/js/hs_file_browser.js (hunk 2 above)
Comment (sblack-usu) [reply theme/static/js/hs_file_browser.js]: yeah, that's a good catch.  Without var it is global.  I'll make the change

File: theme/static/js/hs_file_browser.js [hunk 4]
```diff
@@ -1669,7 +1673,7 @@ $(document).ready(function () {
         // only need that path after /data/contents/
         var path = file.attr("data-url").split('/data/contents/')[1];
         var fullURL;
-        if (~$(this).attr("url_aggregation").indexOf("HS_JS_AGG_KEY")) {
+        if ($(this).attr("url_aggregation")) {
```
Comment (hyi) [inline theme/static/js/hs_file_browser.js]: Should ```indexOf()``` call be retained so that the replacement only happens when it is needed?

Comment (sblack-usu) [reply theme/static/js/hs_file_browser.js]: it's not needed, the replace will just return the string unchanged if that key doesn't.  This change allows urls that have HS_JS_MAIN_FILE_KEY and not HS_JS_AGG_KEY.

**Pull request merged**